### PR TITLE
TCOMP-1920 : active if fix

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-1920

**What is the current behavior?** (You can also link to an open issue here)
Fail if 2 ActiveIf on same target.

**What is the new behavior?**
Ok.

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        def.setPath("hello");
+        def.setMetadata(new HashMap<>());
+        def.getMetadata().put(Metadatas.CONDITION_IF_TARGET, "value");
+
+        final PropertyNode node = new PropertyNode(prop, EParameterFieldType.MAPPING_TYPE, true);
+
+        node.getLayouts().put("Main", new Layout("Main"));
+        return node;
+    }
+}
```
Comment (undx) [inline main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java:56]: I reverted your changes on SettingVisitor and test ran successfully! So, this test is not testing change made nor issue goal.

Comment (undx) [reply main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java:56]: forgot to say, there's no assertions also ;-)

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SettingVisitor.java [hunk 2]
```diff
@@ -188,9 +188,10 @@ private void buildActivationCondition(final PropertyNode node, final PropertyNod
                         }
                     })
                     .map(this.settings::get)
-                    .filter(Objects::nonNull).filter(TaCoKitElementParameter.class::isInstance)
+                    .filter(Objects::nonNull)
+                    .filter(TaCoKitElementParameter.class::isInstance)
                     .map(TaCoKitElementParameter.class::cast)
-                    .collect(toMap(ElementParameter::getName, identity()));
+                    .collect(toMap(ElementParameter::getName, identity(), (a, b) -> a));
```
Comment (undx) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SettingVisitor.java:194]: Your just avoiding to have a same key on the map and an illegalargex. Is it really relevant as stated in issue with activeifs?
Did not really dig into, just asking...

**Review by undx — COMMENTED** [review]
See my comments

File: main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/debounce/DebounceManagerTest.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 package org.talend.sdk.component.studio.debounce;
 
 import static java.lang.Thread.sleep;
-import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
[the remark this answers is not in the document]
Comment (clesaec) [reply main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/debounce/DebounceManagerTest.java:19]: Unit tests didn't work on my IDE ... And others unit test on this project are in Junit 5.
With few change, it avoid JUnit 4.

File: main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java (hunk 1 above)
Comment (clesaec) [reply main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java:56]: Need a call ? i get the exception if i remove lambda ", (a, b) -> a" in code

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SettingVisitor.java (hunk 2 above)
Comment (clesaec) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SettingVisitor.java:194]: Yes, if you see exception in https://jira.talendforge.org/browse/TCOMP-1920
"IllegalStateException: Duplicate key" is what [you can see in java 8 source code](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/stream/Collectors.java#l1212), with [throwingMerger()](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/stream/Collectors.java#l133) method.

File: main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java (hunk 1 above)
Comment (clesaec) [reply main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/SettingVisitorTest.java:56]: I have to add Assertions (basic one)

File: main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/ValueConverterTest.java [hunk 4]
```diff
@@ -40,23 +39,23 @@ void testToTable() {
 
         String table = "[{key1=value11, key2=value12}, {key1=value21, key2=value22}]";
         List<Map<String, Object>> converted = ValueConverter.toTable(table);
-        assertEquals(2, converted.size());
-        assertEquals(expected0, converted.get(0));
-        assertEquals(expected1, converted.get(1));
+        Assertions.assertEquals(2, converted.size());
+        Assertions.assertEquals(expected0, converted.get(0));
+        Assertions.assertEquals(expected1, converted.get(1));
```
[the remark this answers is not in the document]
Comment (clesaec) [reply main/plugins/org.talend.sdk.component.studio-integration/src/test/java/org/talend/sdk/component/studio/model/parameter/ValueConverterTest.java:44]: Same, change JUnit4 in 5 ... elsewhere, it generate error on my IDE.

**Review by clesaec — COMMENTED** [review]
Answers

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SettingVisitor.java (hunk 2 above)
Comment (undx) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/SettingVisitor.java:194]: my fault! 👍🏻

**Review by undx — APPROVED** [review]
LGTM

**Review by dlin-talend — APPROVED** [review]

**Pull request merged**